### PR TITLE
(RSFB-929) REGEXP_LIKE function fix

### PR DESCRIPTION
**Pull request by Kowsikraja**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -1336,7 +1342,14 @@ private SqlCharStringLiteral modifyIfRegexpContainsSecondOperand(SqlCall call) {
...
     case "i":
-      return modifyRegexStringForMatchArgument(call, "(?i)");
+      String updatedRegexForI = removeLeadingAndTrailingSingleQuotes
+          (call.operand(1).toString());
+      if (updatedRegexForI.startsWith("^") && updatedRegexForI.endsWith("$")) {
+        updatedRegexForI = "(?i)".concat(updatedRegexForI);
+      } else {
+        updatedRegexForI = "^(?i)".concat(updatedRegexForI).concat("$");
+      }
+      return SqlLiteral.createCharString(updatedRegexForI, SqlParserPos.ZERO);
```
Comment (nosheenSheikh02) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:1352]: Wrap this logic into a method. 1345:1352

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -11383,9 +11383,9 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
         .ok(expectedBQ);
   }
 
-  @Test public void testForRegexpLikeFunction() {
+  @Test public void testForRegexpSimilarFunction() {
     final RelBuilder builder = relBuilder();
-    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_LIKE,
+    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_SIMILAR,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11388]: make it regexp_similar => regexpSimilar

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -11403,9 +11403,9 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.SPARK.getDialect()), isLinux(expectedSparkSql));
   }
 
-  @Test public void testForRegexpLikeFunctionWithThirdArgumentAsI() {
+  @Test public void testForRegexpSimilarFunctionWithThirdArgumentAsI() {
     final RelBuilder builder = relBuilder();
-    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_LIKE,
+    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_SIMILAR,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11408]: make it regexp_similar => regexpSimilar

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 4]
```diff
@@ -11424,9 +11424,9 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.SPARK.getDialect()), isLinux(expectedSparkSql));
   }
 
-  @Test public void testForRegexpLikeFunctionWithThirdArgumentAsX() {
+  @Test public void testForRegexpSimilarFunctionWithThirdArgumentAsX() {
     final RelBuilder builder = relBuilder();
-    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_LIKE,
+    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_SIMILAR,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11429]: make it regexp_similar => regexpSimilar

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 5]
```diff
@@ -11444,9 +11444,9 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
   }
 
 
-  @Test public void testForRegexpLikeFunctionWithThirdArgumentAsC() {
+  @Test public void testForRegexpSimilarFunctionWithThirdArgumentAsC() {
     final RelBuilder builder = relBuilder();
-    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_LIKE,
+    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_SIMILAR,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11449]: make it regexp_similar => regexpSimilar

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 6]
```diff
@@ -11465,9 +11465,9 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.SPARK.getDialect()), isLinux(expectedSparkSql));
   }
 
-  @Test public void testForRegexpLikeFunctionWithThirdArgumentAsN() {
+  @Test public void testForRegexpSimilarFunctionWithThirdArgumentAsN() {
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11468]: make it regexp_similar => regexpSimilar

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 7]
```diff
@@ -11482,9 +11482,26 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
     assertThat(toSql(root, DatabaseProduct.SPARK.getDialect()), isLinux(expectedSparkSql));
   }
 
-  @Test public void testForRegexpLikeFunctionWithThirdArgumentAsM() {
+  @Test public void testForRegexpLikeFunctionWithThirdArgumentAsI() {
+    final RelBuilder builder = relBuilder();
+    final RexNode regexp_like = builder.call(SqlLibraryOperators.REGEXP_LIKE,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11487]: make it regexp_like => regexpLike

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 8]
```diff
@@ -11482,9 +11482,26 @@ RelNode createLogicalValueRel(RexNode col1, RexNode col2) {
...
+        + "r'^(?i)Mike B(i|y)RD$') AS A\n"
+        + "FROM scott.EMP";
+
+    assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBqSql));
+  }
+
+  @Test public void testForRegexpSimilarFunctionWithThirdArgumentAsM() {
     final RelBuilder builder = relBuilder();
-    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_LIKE,
+    final RexNode regexp_similar = builder.call(SqlLibraryOperators.REGEXP_SIMILAR,
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:11504]: make it regexp_similar => regexpSimilar

**Review by nosheenSheikh02 — CHANGES_REQUESTED** [review]

**Comment by nosheenSheikh02** [comment]
Please check one test case is failing in ReltoSqlConverterTest
`testFunctionsWithRegexOperands`

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 2 above)
Comment (Kowsikraja) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: I had resolved that please have a look

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 3 above)
Comment (Kowsikraja) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: i had resolved that comments please have a look

**Comment by Kowsikraja** [comment]
> Please check one test case is failing in ReltoSqlConverterTest `testFunctionsWithRegexOperands`

i have changed that please have a look

**Comment by nosheenSheikh02** [comment]
> > Please check one test case is failing in ReltoSqlConverterTest `testFunctionsWithRegexOperands`
> 
> i have changed that please have a look

Okay LGTM

**Review by nosheenSheikh02 — APPROVED** [review]

**Review by dm-shantanuc — APPROVED** [review]
Changes look good. Approved by Anurag too.

**Pull request merged**